test/test_loaded_meshes.cpp

```diff
@@ -0,0 +1,254 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2008, Willow Garage, Inc.
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Willow Garage nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/** \Author Jorge Santos */
+
+#include <geometric_shapes/bodies.h>
+#include <geometric_shapes/shape_operations.h>
+#include <geometric_shapes/body_operations.h>
+#include <boost/filesystem.hpp>
+#include <gtest/gtest.h>
+#include "resources/config.h"
+
+
+/**
+ * Test fixture that generates meshes from the primitive shapes SPHERE, CYLINDER, CONE and BOX,
+ * and load their twins from STL files. All the following tests are intended to verify that both
+ * procedures produce equivalent meshes, and particularly that changes related to issue #38 don't
+ * break mesh loading.
+ */
+class CompareMeshVsPrimitive : public ::testing::Test
+{
+public:
+  CompareMeshVsPrimitive()
+  {
+  }
+
+  void SetUp()
+  {
+    // WARN1: we skip the sphere because it takes ages for each test, around 20 seconds each
+    // (possible bug?)
```

Comment:
Please create an issue with a simple test-case so that we can reproduce and analyze it.
There is no sense in adding warnings about possible library bugs into the code of the library...